docs/examples/plot_intermediate_outputs.py

```diff
@@ -0,0 +1,268 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+Walk through intermediate outputs
+=================================
+
+We reuse the example :ref:`example-complex-pipeline` and
+walk through intermediates outputs. It is very likely a converted
+model gives different outputs or fails due to a custom
+converter which is not correctly implemented.
+One option is to look into the output of every node of the
+ONNX graph.
+
+
+.. contents::
+    :local:
+
+Create and train a complex pipeline
++++++++++++++++++++++++++++++++++++
+
+We reuse the pipeline implemented in example
+`Column Transformer with Mixed Types
+<https://scikit-learn.org/stable/auto_examples/compose/plot_column_transformer_mixed_types.html#sphx-glr-auto-examples-compose-plot-column-transformer-mixed-types-py>`_.
+There is one change because
+`ONNX-ML Imputer <https://github.com/onnx/onnx/blob/master/docs/Operators-ml.md#ai.onnx.ml.Imputer>`_
+does not handle string type. This cannot be part of the final ONNX pipeline
+and must be removed. Look for comment starting with ``---`` below.
+"""
+import pandas as pd
+import numpy as np
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import Pipeline
+from sklearn.impute import SimpleImputer
+from sklearn.preprocessing import StandardScaler, OneHotEncoder
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+
+titanic_url = ('https://raw.githubusercontent.com/amueller/'
+               'scipy-2017-sklearn/091d371/notebooks/datasets/titanic3.csv')
+data = pd.read_csv(titanic_url)
+X = data.drop('survived', axis=1)
+y = data['survived']
+
+# SimpleImputer on string is not available for string in ONNX-ML specifications.
+# So we do it beforehand.
+for cat in ['embarked', 'sex', 'pclass']:
+    X[cat].fillna('missing', inplace=True)
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
+
+numeric_features = ['age', 'fare']
+numeric_transformer = Pipeline(steps=[
+    ('imputer', SimpleImputer(strategy='median')),
+    ('scaler', StandardScaler())])
+
+categorical_features = ['embarked', 'sex', 'pclass']
+categorical_transformer = Pipeline(steps=[
+    # --- SimpleImputer is not available for strings in ONNX-ML specifications. 
+    # ('imputer', SimpleImputer(strategy='constant', fill_value='missing')),
+    ('onehot', OneHotEncoder(handle_unknown='ignore'))])
+
+preprocessor = ColumnTransformer(
+    transformers=[
+        ('num', numeric_transformer, numeric_features),
+        ('cat', categorical_transformer, categorical_features),
+        ])
+
+clf = Pipeline(steps=[('preprocessor', preprocessor),
+                      ('classifier', LogisticRegression(solver='lbfgs'))])
+
```

Comment:
There should be zero or one blank line here. Two blank lines should be used to surround top-level functions and class definitions.